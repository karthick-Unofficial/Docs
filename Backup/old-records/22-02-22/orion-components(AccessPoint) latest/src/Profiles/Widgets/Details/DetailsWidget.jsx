import React, { PureComponent } from "react";
import { withSpan } from "../../../Apm"; 
import PropTypes from "prop-types";
import { timeConversion } from "client-app-core";
import { SimpleTable } from "../../../CBComponents";
import _ from "lodash";
import { BaseWidget } from "../shared";
import VisualDetail from "./components/VisualDetail";
import { Grid, Collapse, Button } from "@material-ui/core";
import { ExpandLess, ExpandMore } from "@material-ui/icons";
import { Translate, getTranslation } from "orion-components/i18n/I18nContainer";

const propTypes = {
	order: PropTypes.number,
	details: PropTypes.object.isRequired,
	displayProps: PropTypes.array,
	dir: PropTypes.string
};

const defaultProps = {
	order: 0,
	displayProps: null,
	dir: "ltr"
};

export class DetailsWidget extends PureComponent {
	constructor(props) {
		super(props);
		const { displayProps } = this.props;
		this.state = {
			expanded: false,
			hasVisuals: !!displayProps
		};
	}

	componentDidMount() {
		const { displayProps, details } = this.props;
		if (displayProps) {
			const hasVisuals = displayProps.find(prop => {
				const { visual, key } = prop;
				return visual && details[key];
			});
			this.setState({ hasVisuals: !!hasVisuals });
		}
	}

	getSimpleDetails = () => {
		const { details, displayProps, timeFormatPreference } = this.props;
		let rows = [];
		if (displayProps) {
			const simpleProps = displayProps.filter(prop => {
				const { visual, key } = prop;
				return !visual && (details[key] || typeof details[key] === "number") && details[key] !== "";
			});
			rows = simpleProps.map(prop => {
				const { key, label, unit } = prop;
				let value = details[key];
				if (unit) {
					switch (unit) {
						case "time":
							value = timeConversion.convertToUserTime(details[key], `full_${timeFormatPreference}`);
							break;
						default:
							value = details[key];
							break;
					}
				}
				return {
					label,
					unit,
					value
				};
			});
		} else {
			rows = _.map(_.pickBy(details, detail => !!detail), (value, label) => {
				if (value) {
					return { label, value };
				}
			});
		}

		return <SimpleTable rows={rows} />;
	};

	getVisualDetails = () => {
		const { details, displayProps } = this.props;
		const visualProps = displayProps.filter(prop => {
			const { visual, key, tooltip } = prop;
			return (
				visual &&
				(details[key] || typeof details[key] === "number" || details[tooltip]) &&
				details[key] !== ""
			);
		});
		const visuals = visualProps.map(prop => {
			const { key, label, tooltip, unit, visual } = prop;
			return (
				<Grid key={key} item>
					<VisualDetail
						label={label}
						tooltip={details[tooltip]}
						unit={unit}
						value={details[key]}
						visual={visual}
					/>
				</Grid>
			);
		});
		return visuals;
	};

	handleExpand = () => {
		const { expanded } = this.state;
		this.setState({
			expanded: !expanded
		});
	};

	render() {
		const { order, enabled, dir } = this.props;
		const { expanded, hasVisuals } = this.state;
		return (
			<BaseWidget enabled={enabled} order={order} title={getTranslation("global.profiles.widgets.details.title")} dir={dir}>
				{hasVisuals && (
					<Grid
						container
						style={{ margin: "10px -12px" }}
						justify="space-around"
						spacing={3}
						alignContent="center"
					>
						{this.getVisualDetails()}
					</Grid>
				)}
				<Collapse
					className="list-body-wrapper"
					style={{ overflowX: "scroll" }}
					in={expanded || !hasVisuals}
				>
					{this.getSimpleDetails()}
				</Collapse>
				{hasVisuals && (
					<Button onClick={this.handleExpand}>
						{expanded ? <Translate value="global.profiles.widgets.details.showLess"/> : <Translate value="global.profiles.widgets.details.showMore"/>}
						{expanded ? <ExpandLess /> : <ExpandMore />}
					</Button>
				)}
			</BaseWidget>
		);
	}
}

DetailsWidget.propTypes = propTypes;
DetailsWidget.defaultProps = defaultProps;

export default withSpan("details-widget", "profile-widget")(DetailsWidget);
