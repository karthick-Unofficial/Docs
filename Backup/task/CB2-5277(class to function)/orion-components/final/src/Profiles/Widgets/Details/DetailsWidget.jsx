import React, { useEffect, useState } from "react";
import { withSpan } from "../../../Apm";
import PropTypes from "prop-types";
import { timeConversion } from "client-app-core";
import { SimpleTable } from "../../../CBComponents";
import _ from "lodash";
import { BaseWidget } from "../shared";
import VisualDetail from "./components/VisualDetail";
import { Grid, Collapse, Button } from "@material-ui/core";
import { ExpandLess, ExpandMore } from "@material-ui/icons";
import { Translate, getTranslation } from "orion-components/i18n/I18nContainer";

const propTypes = {
	order: PropTypes.number,
	details: PropTypes.object.isRequired,
	displayProps: PropTypes.array,
	dir: PropTypes.string
};

const defaultProps = {
	order: 0,
	displayProps: null,
	dir: "ltr"
};

const DetailsWidget = ({
	displayProps,
	details,
	timeFormatPreference,
	dir,
	order,
	enabled
}) => {

	const [expanded, setExpanded] = useState(false);
	const [hasVisuals, setHasVisuals] = useState(!!displayProps);

	useEffect(() => {
		if (displayProps) {
			const hasVisuals = displayProps.find(prop => {
				const { visual, key } = prop;
				return visual && details[key];
			});
			setHasVisuals(!!hasVisuals);
		}
	}, []);

	const getSimpleDetails = () => {
		let rows = [];
		if (displayProps) {
			const simpleProps = displayProps.filter(prop => {
				const { visual, key } = prop;
				return !visual && (details[key] || typeof details[key] === "number") && details[key] !== "";
			});
			rows = simpleProps.map(prop => {
				const { key, label, unit } = prop;
				let value = details[key];
				if (unit) {
					switch (unit) {
						case "time":
							value = timeConversion.convertToUserTime(details[key], `full_${timeFormatPreference}`);
							break;
						default:
							value = details[key];
							break;
					}
				}
				return {
					label,
					unit,
					value
				};
			});
		} else {
			rows = _.map(_.pickBy(details, detail => !!detail), (value, label) => {
				if (value) {
					return { label, value };
				}
			});
		}

		return <SimpleTable rows={rows} dir={dir} />;
	};

	const getVisualDetails = () => {
		const visualProps = displayProps.filter(prop => {
			const { visual, key, tooltip } = prop;
			return (
				visual &&
				(details[key] || typeof details[key] === "number" || details[tooltip]) &&
				details[key] !== ""
			);
		});
		const visuals = visualProps.map(prop => {
			const { key, label, tooltip, unit, visual } = prop;
			return (
				<Grid key={key} item>
					<VisualDetail
						label={label}
						tooltip={details[tooltip]}
						unit={unit}
						value={details[key]}
						visual={visual}
					/>
				</Grid>
			);
		});
		return visuals;
	};

	const handleExpand = () => {
		setExpanded(!expanded);
	};

	return (
		<BaseWidget enabled={enabled} order={order} title={getTranslation("global.profiles.widgets.details.title")} dir={dir}>
			{hasVisuals && (
				<Grid
					container
					style={{ margin: "10px -12px" }}
					justifyContent="space-around"
					spacing={3}
					alignContent="center"
				>
					{getVisualDetails()}
				</Grid>
			)}
			<Collapse
				className="list-body-wrapper"
				style={{ overflowX: "scroll" }}
				in={expanded || !hasVisuals}
			>
				{getSimpleDetails()}
			</Collapse>
			{hasVisuals && (
				<Button onClick={handleExpand}>
					{expanded ? <Translate value="global.profiles.widgets.details.showLess" /> : <Translate value="global.profiles.widgets.details.showMore" />}
					{expanded ? <ExpandLess /> : <ExpandMore />}
				</Button>
			)}
		</BaseWidget>
	);
};

DetailsWidget.propTypes = propTypes;
DetailsWidget.defaultProps = defaultProps;

export default withSpan("details-widget", "profile-widget")(DetailsWidget);
