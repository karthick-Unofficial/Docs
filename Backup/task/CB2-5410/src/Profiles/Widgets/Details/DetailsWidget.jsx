import React, { useEffect, useState } from "react";
import { withSpan } from "../../../Apm";
import PropTypes from "prop-types";
import { timeConversion } from "client-app-core";
import { SimpleTable } from "../../../CBComponents";
import { BaseWidget } from "../shared";
import VisualDetail from "./components/VisualDetail";
import { Grid, Collapse, Button } from "@mui/material";
import { ExpandLess, ExpandMore } from "@mui/icons-material";
import { Translate, getTranslation } from "orion-components/i18n";
import map from "lodash/map";
import pickBy from "lodash/pickBy";
import { useSelector } from "react-redux";
import { getWidgetState } from "orion-components/Profiles/Selectors";
import { getDir } from "orion-components/i18n/Config/selector";
import WidgetMenu from "./components/WidgetMenu";
import { getGlobalWidgetState } from "../Selectors";

const propTypes = {
	id: PropTypes.string,
	details: PropTypes.object.isRequired,
	displayProps: PropTypes.array
};

const defaultProps = {
	displayProps: null
};

const widgetName = "detailsWidget";

const DetailsWidget = ({ displayProps, details, id }) => {
	const enabled = useSelector((state) => getWidgetState(state)(id, "enabled"));
	const widgetState = useSelector((state) => getGlobalWidgetState(state)(widgetName)) || {};
	const expandedState = widgetState?.autoExpand;
	const dir = useSelector((state) => getDir(state));

	const timeFormatPreference = useSelector((state) => state?.appState?.global?.timeFormat) || "12-hour";

	const [expanded, setExpanded] = useState(expandedState);
	const [hasVisuals, setHasVisuals] = useState(!!displayProps);

	useEffect(() => {
		if (displayProps) {
			const hasVisuals = displayProps.find((prop) => {
				const { visual, key } = prop;
				return visual && details[key];
			});
			setHasVisuals(!!hasVisuals);
		}
	}, []);

	const getSimpleDetails = () => {
		let rows = [];
		if (displayProps) {
			const simpleProps = displayProps.filter((prop) => {
				const { visual, key } = prop;
				return !visual && (details[key] || typeof details[key] === "number") && details[key] !== "";
			});
			rows = simpleProps.map((prop) => {
				const { key, label, unit } = prop;
				let value = details[key];
				if (unit) {
					switch (unit) {
						case "time":
							value = timeConversion.convertToUserTime(details[key], `full_${timeFormatPreference}`);
							break;
						default:
							value = details[key];
							break;
					}
				}
				return {
					label,
					unit,
					value
				};
			});
		} else {
			rows = map(
				pickBy(details, (detail) => !!detail),
				(value, label) => {
					if (value) {
						return { label, value };
					}
				}
			);
		}

		return <SimpleTable rows={rows} dir={dir} />;
	};

	const getVisualDetails = () => {
		const visualProps = displayProps.filter((prop) => {
			const { visual, key, tooltip } = prop;
			return (
				visual && (details[key] || typeof details[key] === "number" || details[tooltip]) && details[key] !== ""
			);
		});
		const visuals = visualProps.map((prop) => {
			const { key, label, tooltip, unit, visual } = prop;
			return (
				<Grid key={key} item>
					<VisualDetail
						label={label}
						tooltip={details[tooltip]}
						unit={unit}
						value={details[key]}
						visual={visual}
					/>
				</Grid>
			);
		});
		return visuals;
	};

	const handleExpand = () => {
		setExpanded(!expanded);
	};

	const getWidgetMenu = () => {
		return <WidgetMenu
			widgetName={widgetName}
			widgetState={widgetState}
			expandedState={expandedState}
			dir={dir}
		/>
	};

	return (
		<BaseWidget
			enabled={enabled}
			title={getTranslation("global.profiles.widgets.details.title")}
			dir={dir}
			widgetMenu={getWidgetMenu()}
		>
			{hasVisuals && (
				<Grid
					container
					style={{ margin: "10px -12px" }}
					justifyContent="space-around"
					spacing={3}
					alignContent="center"
				>
					{getVisualDetails()}
				</Grid>
			)}
			<Collapse className="list-body-wrapper" style={{ overflowX: "scroll" }} in={expanded || !hasVisuals}>
				{getSimpleDetails()}
			</Collapse>
			{hasVisuals && (
				<Button onClick={handleExpand}>
					{expanded ? (
						<Translate value="global.profiles.widgets.details.showLess" />
					) : (
						<Translate value="global.profiles.widgets.details.showMore" />
					)}
					{expanded ? <ExpandLess /> : <ExpandMore />}
				</Button>
			)}
		</BaseWidget>
	);
};

DetailsWidget.propTypes = propTypes;
DetailsWidget.defaultProps = defaultProps;

export default withSpan("details-widget", "profile-widget")(DetailsWidget);
