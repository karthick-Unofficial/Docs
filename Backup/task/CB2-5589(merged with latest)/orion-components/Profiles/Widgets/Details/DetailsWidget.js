"use strict";

var _typeof = require("@babel/runtime-corejs3/helpers/typeof");
var _WeakMap = require("@babel/runtime-corejs3/core-js-stable/weak-map");
var _Object$defineProperty = require("@babel/runtime-corejs3/core-js-stable/object/define-property");
var _Object$getOwnPropertyDescriptor = require("@babel/runtime-corejs3/core-js-stable/object/get-own-property-descriptor");
var _interopRequireDefault = require("@babel/runtime-corejs3/helpers/interopRequireDefault");
_Object$defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _find = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/find"));
var _filter = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/filter"));
var _map = _interopRequireDefault(require("@babel/runtime-corejs3/core-js-stable/instance/map"));
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime-corejs3/helpers/slicedToArray"));
var _react = _interopRequireWildcard(require("react"));
var _Apm = require("../../../Apm");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _clientAppCore = require("client-app-core");
var _CBComponents = require("../../../CBComponents");
var _shared = require("../shared");
var _VisualDetail = _interopRequireDefault(require("./components/VisualDetail"));
var _material = require("@mui/material");
var _iconsMaterial = require("@mui/icons-material");
var _i18n = require("orion-components/i18n");
var _map2 = _interopRequireDefault(require("lodash/map"));
var _pickBy = _interopRequireDefault(require("lodash/pickBy"));
function _getRequireWildcardCache(nodeInterop) { if (typeof _WeakMap !== "function") return null; var cacheBabelInterop = new _WeakMap(); var cacheNodeInterop = new _WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }
function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = _Object$defineProperty && _Object$getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? _Object$getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { _Object$defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
var propTypes = {
  order: _propTypes["default"].number,
  details: _propTypes["default"].object.isRequired,
  displayProps: _propTypes["default"].array,
  dir: _propTypes["default"].string
};
var defaultProps = {
  order: 0,
  displayProps: null,
  dir: "ltr"
};
var DetailsWidget = function DetailsWidget(_ref) {
  var displayProps = _ref.displayProps,
    details = _ref.details,
    timeFormatPreference = _ref.timeFormatPreference,
    dir = _ref.dir,
    order = _ref.order,
    enabled = _ref.enabled;
  var _useState = (0, _react.useState)(false),
    _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
    expanded = _useState2[0],
    setExpanded = _useState2[1];
  var _useState3 = (0, _react.useState)(!!displayProps),
    _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
    hasVisuals = _useState4[0],
    setHasVisuals = _useState4[1];
  (0, _react.useEffect)(function () {
    if (displayProps) {
      var _hasVisuals = (0, _find["default"])(displayProps).call(displayProps, function (prop) {
        var visual = prop.visual,
          key = prop.key;
        return visual && details[key];
      });
      setHasVisuals(!!_hasVisuals);
    }
  }, []);
  var getSimpleDetails = function getSimpleDetails() {
    var rows = [];
    if (displayProps) {
      var simpleProps = (0, _filter["default"])(displayProps).call(displayProps, function (prop) {
        var visual = prop.visual,
          key = prop.key;
        return !visual && (details[key] || typeof details[key] === "number") && details[key] !== "";
      });
      rows = (0, _map["default"])(simpleProps).call(simpleProps, function (prop) {
        var key = prop.key,
          label = prop.label,
          unit = prop.unit;
        var value = details[key];
        if (unit) {
          switch (unit) {
            case "time":
              value = _clientAppCore.timeConversion.convertToUserTime(details[key], "full_".concat(timeFormatPreference));
              break;
            default:
              value = details[key];
              break;
          }
        }
        return {
          label: label,
          unit: unit,
          value: value
        };
      });
    } else {
      rows = (0, _map2["default"])((0, _pickBy["default"])(details, function (detail) {
        return !!detail;
      }), function (value, label) {
        if (value) {
          return {
            label: label,
            value: value
          };
        }
      });
    }
    return /*#__PURE__*/_react["default"].createElement(_CBComponents.SimpleTable, {
      rows: rows,
      dir: dir
    });
  };
  var getVisualDetails = function getVisualDetails() {
    var visualProps = (0, _filter["default"])(displayProps).call(displayProps, function (prop) {
      var visual = prop.visual,
        key = prop.key,
        tooltip = prop.tooltip;
      return visual && (details[key] || typeof details[key] === "number" || details[tooltip]) && details[key] !== "";
    });
    var visuals = (0, _map["default"])(visualProps).call(visualProps, function (prop) {
      var key = prop.key,
        label = prop.label,
        tooltip = prop.tooltip,
        unit = prop.unit,
        visual = prop.visual;
      return /*#__PURE__*/_react["default"].createElement(_material.Grid, {
        key: key,
        item: true
      }, /*#__PURE__*/_react["default"].createElement(_VisualDetail["default"], {
        label: label,
        tooltip: details[tooltip],
        unit: unit,
        value: details[key],
        visual: visual
      }));
    });
    return visuals;
  };
  var handleExpand = function handleExpand() {
    setExpanded(!expanded);
  };
  return /*#__PURE__*/_react["default"].createElement(_shared.BaseWidget, {
    enabled: enabled,
    order: order,
    title: (0, _i18n.getTranslation)("global.profiles.widgets.details.title"),
    dir: dir
  }, hasVisuals && /*#__PURE__*/_react["default"].createElement(_material.Grid, {
    container: true,
    style: {
      margin: "10px -12px"
    },
    justifyContent: "space-around",
    spacing: 3,
    alignContent: "center"
  }, getVisualDetails()), /*#__PURE__*/_react["default"].createElement(_material.Collapse, {
    className: "list-body-wrapper",
    style: {
      overflowX: "scroll"
    },
    "in": expanded || !hasVisuals
  }, getSimpleDetails()), hasVisuals && /*#__PURE__*/_react["default"].createElement(_material.Button, {
    onClick: handleExpand
  }, expanded ? /*#__PURE__*/_react["default"].createElement(_i18n.Translate, {
    value: "global.profiles.widgets.details.showLess"
  }) : /*#__PURE__*/_react["default"].createElement(_i18n.Translate, {
    value: "global.profiles.widgets.details.showMore"
  }), expanded ? /*#__PURE__*/_react["default"].createElement(_iconsMaterial.ExpandLess, null) : /*#__PURE__*/_react["default"].createElement(_iconsMaterial.ExpandMore, null)));
};
DetailsWidget.propTypes = propTypes;
DetailsWidget.defaultProps = defaultProps;
var _default = (0, _Apm.withSpan)("details-widget", "profile-widget")(DetailsWidget);
exports["default"] = _default;