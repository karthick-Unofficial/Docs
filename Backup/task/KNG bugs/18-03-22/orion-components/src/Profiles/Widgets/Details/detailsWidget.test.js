import React from "react";
import DetailsWidget from "./DetailsWidget";

import { shallow } from "enzyme";

jest.mock("mapbox-gl/dist/mapbox-gl", () => ({
	Map: () => ({})
}));

describe("DetailsWidget", () => {
	it("renders", () => {
		const selectedEntity = {
			appId: "vessels-app",
			entityData: {
				geometry: {
					coordinates: [-91.19507833333333, 29.682976666666665],
					type: "Point"
				},
				properties: {
					course: 209.3,
					disposition: "unknown",
					name: "MISS LAURIE",
					sourceId: "367002040",
					speed: 7.2
				}
			},
			entityType: "track",
			feedId: "aishub",
			id: "aishub.367002040",
			isActive: true,
			ownerOrg: "ares_security_corporation",
			sourceId: "367002040"
		};

		const wrapper = shallow(
			<DetailsWidget
				selectedEntity={selectedEntity}
				details={selectedEntity.entityData.properties}
				displayProps={[]}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
