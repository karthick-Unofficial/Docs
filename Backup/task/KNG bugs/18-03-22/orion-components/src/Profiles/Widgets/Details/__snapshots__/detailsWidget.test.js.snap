// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsWidget renders 1`] = `
<WithStyles(BaseWidget)
  order={0}
  title="Details"
>
  <WithStyles(ForwardRef(Collapse))
    className="list-body-wrapper"
    in={true}
    style={
      Object {
        "overflowX": "scroll",
      }
    }
  >
    <WithStyles(SimpleTable)
      rows={Array []}
    />
  </WithStyles(ForwardRef(Collapse))>
</WithStyles(BaseWidget)>
`;
